### PR TITLE
Swipe Navigation

### PR DESCRIPTION
**Pull request by wojtus7**
New navigation is based on [Material Top Tabs Navigator](https://reactnavigation.org/docs/material-top-tab-navigator/)

Since all screens are swipeable we don't mount/unmount them anymore so we lost some options used previously but if we want we can lazy load them.
Also `reset` function worked bad with swipe animation so I replaced it `navigate` in `useBackToHome` func and added additional `useResetToHome` where necessary

From what I tested all swipe directions works together on home screen (We use all direction now for navigation, pull refresh and bottom sheet)

Maybe I don't know about some screens or interaction so I encourage to test app navigation heavily 🫠

iOS:

https://github.com/daimo-eth/daimo/assets/42337257/8c784fc0-66a3-4b6e-94a0-e0814aa84ce1

Issue on android is I had to disable animations when navigate using buttons because it was breaking text focus and keyboard insteractions (Known issue you can read more about eg [here](https://github.com/react-navigation/react-navigation/issues/10397) or [here](https://stackoverflow.com/questions/74656402/material-top-tabs-closes-keyboard-on-first-focus)

https://github.com/daimo-eth/daimo/assets/42337257/da1e8611-b89a-4115-a786-2180f910994f

### CODE REVIEW

File: apps/daimo-mobile/package.json [hunk 1]
```diff
@@ -72,21 +74,22 @@
     "jest-expo": "^49.0.0",
     "react": "18.2.0",
     "react-native": "0.72.4",
+    "react-native-gesture-handler": "~2.12.0",
     "react-native-get-random-values": "~1.9.0",
     "react-native-mmkv": "^2.8.0",
+    "react-native-pager-view": "^6.2.2",
     "react-native-qrcode-svg": "^6.2.0",
+    "react-native-reanimated": "~3.3.0",
```
Comment (dcposch) [inline apps/daimo-mobile/package.json:82]: :rocket:

File: apps/daimo-mobile/src/view/TabNav.tsx [hunk 2]
```diff
@@ -66,25 +59,25 @@ export function TabNav() {
...
-      <Tab.Screen name="DepositTab" component={DepositTab} options={unmount} />
-      <Tab.Screen name="ReceiveTab" component={ReceiveTab} options={unmount} />
-      <Tab.Screen name="HomeTab" component={HomeTab} options={opts} />
-      <Tab.Screen name="SendTab" component={SendTab} options={unmount} />
-      <Tab.Screen name="SettingsTab" component={SettingsTab} options={opts} />
+      <Tab.Screen name="DepositTab" component={DepositTab} />
+      <Tab.Screen name="ReceiveTab" component={ReceiveTab} />
+      <Tab.Screen name="HomeTab" component={HomeTab} />
+      <Tab.Screen name="SendTab" component={SendTab} />
+      <Tab.Screen name="SettingsTab" component={SettingsTab} />
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/TabNav.tsx:68]: 👍

File: apps/daimo-mobile/src/view/screen/receive/ReceiveScreen.tsx [hunk 3]
```diff
@@ -46,9 +47,28 @@ function RequestScreenInner({
...
+    let focusTimeout;
+    if (isFocused && autoFocus && Platform.OS === "ios") {
+      const { setParams }: { setParams: any } = nav;
+      setParams({
+        autoFocus: false,
+      });
+      // wait for the screen transition animation to finish before open keyboard
+      focusTimeout = setTimeout(() => {
+        textInputRef.current?.focus();
+      }, 500);
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/receive/ReceiveScreen.tsx:64]: do the screen transitions take 0.5sec?

File: apps/daimo-mobile/src/view/shared/ScreenHeader.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ export function ScreenHeader({
   );
 }
 
-export function useExitToHome() {
+export function useResetToHome() {
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/shared/ScreenHeader.tsx:35]: 👍

**Review by dcposch — APPROVED** [review]
Looks great to me

File: apps/daimo-mobile/src/view/screen/receive/ReceiveScreen.tsx (hunk 3 above)
Comment (wojtus7) [reply apps/daimo-mobile/src/view/screen/receive/ReceiveScreen.tsx:64]: It takes 300ms but I rather take a bufor if anything loads or something happens on JS thread that would delay the animation.
Afraid of living on the edge 🫠

**Pull request merged**